### PR TITLE
Fix command: Migrate v1 Setters to latest

### PR DESCRIPTION
**Pull request by phanimarupaka**
@mortent @pwittrock 

This PR is to upgrade setters from V1 to latest. All the v1 setter comments are converted to latest setters and substitutions.

### CODE REVIEW

File: kyaml/go.mod [hunk 1]
```diff
@@ -11,11 +11,11 @@ require (
...
-	github.com/spf13/pflag v1.0.3
+	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.4.0
 	github.com/xlab/treeprint v0.0.0-20181112141820-a009c3971eca
 	go.starlark.net v0.0.0-20200306205701-8dd3e2ee1dd5
-	golang.org/x/net v0.0.0-20200226121028-0de0cce0169b // indirect
 	gopkg.in/yaml.v2 v2.3.0
-	gopkg.in/yaml.v3 v3.0.0-20191120175047-4206685974f2
+	gopkg.in/yaml.v3 v3.0.0-20200121175148-a6ecf24a6d71
+	sigs.k8s.io/kustomize/cmd/config v0.4.2
```
Comment (mortent) [inline kyaml/go.mod:20]: We can't have this dependency here. `cmd/config` has dependencies on k8s libraries, and that is not allowed in `kyaml`.

File: kyaml/setters2/settersutil/settercreator.go [hunk 2]
```diff
@@ -43,6 +44,16 @@ type SetterCreator struct {
 	// live apply/preview. This field is added to the setter definition to record
 	// the package publisher's intent to make the setter required to be set.
 	Required bool
+
+	// Path to openAPI file
+	OpenAPIPath string
+
+	// Path to resources folder
+	ResourcesPath string
```
Comment (mortent) [inline kyaml/setters2/settersutil/settercreator.go:52]: Why are these changes needed?

Comment (phanimarupaka) [reply kyaml/setters2/settersutil/settercreator.go:52]: To treat it as a library which can be called from elsewhere apart from cmd/config. Currently, the kyaml libraries for cfg are tightly coupled to cmd/config. This makes it reusable from other places as well. This make this api layer reusable.

File: kyaml/fix/fixsetters/fixsetters.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/setters2"
+	"sigs.k8s.io/kustomize/kyaml/setters2/settersutil"
+)
+
+// SetterFixer fixes setters in the input package
+type SetterFixer struct {
+	// Writer is the stdout to write to
+	Writer io.Writer
```
Comment (mortent) [inline kyaml/fix/fixsetters/fixsetters.go:19]: I don't think this is the right way to do this. We should return a datastructure with the information and then leave it up to the caller to determine how it should be printed. Currently it ends up as something in between a cli-tool and a library.

Comment (phanimarupaka) [reply kyaml/fix/fixsetters/fixsetters.go]: @mortent Used a buffer to return the out messages. Does this look good ?

File: kyaml/go.mod (hunk 1 above)
Comment (phanimarupaka) [reply kyaml/go.mod]: Removed.

File: kyaml/fix/fixsetters/fixsetters.go (hunk 3 above)
Comment (mortent) [reply kyaml/fix/fixsetters/fixsetters.go]: So changing it to a buffer is the same thing, buffer implements the `io.Writer` interface. 
I think we should try to return a struct here that contains the result of fixing the setters as proper data. And it should be returned from the `FixSetters` function. This will allow other tools than a cli to use this functionality. In the current form, it can't be easily used since the caller would have to parse the output to understand what happened.

File: cmd/config/internal/commands/cmdset.go [hunk 4]
```diff
@@ -75,6 +75,11 @@ func initSetterVersion(c *cobra.Command, args []string) error {
 		setterVersion = "v1"
 	}
 
+	if setterVersion == "v1" {
+		fmt.Printf("\nWarning: You are using older version of setters which will be deprecated soon, " +
```
Comment (mortent) [inline cmd/config/internal/commands/cmdset.go:79]: As far as I can see, we are not actually adding the pkg fix command in this PR. We should add this warning when we have the actual command in place.

Comment (phanimarupaka) [reply cmd/config/internal/commands/cmdset.go:79]: I will have the follow up PR for the command in kpt once this is merged so that I can point to kyaml master in kpt. We can release kpt after the command is merged.
